hpx/parallel/util/loop.hpp

```diff
@@ -96,23 +281,98 @@ namespace hpx { namespace parallel { namespace util
                 return it;
             }
         };
+
+        template <typename T>
+        struct loop_n <prefetching_iterator<T>>
+        {
+            /////////////////////////////////////////////////////////////////
+            // handle sequences of non-futures when using prefetching
+            template <typename F>
+            static prefetching_iterator<T> call(prefetching_iterator<T> it,
+                std::size_t count, F && f)
+            {
+                std::size_t cnt = count + 1;
+                for (/**/; cnt != 0; (void) --cnt, ++it)
+                {
+                    std::vector<std::size_t>::iterator inner_it = it.base;
+                    std::size_t j = it.idx;
+                    std::size_t last = it.idx+it.chunk_size;
+
+                    //based on HPX Inspection Report,
+                    //using std::min violates boost min/max
+                    if (it.range_size < last)
```

Comment:
What the inspect was trying to tell you is that you shouldn't use `std::min(a, b)` but write it as `(std::min)(a, b)`. This prevents `min()` from being macro expanded (which happens on some platforms, unfortunately).